### ISSUE TITLE
Possible problem extracting variable fromfor(... as list( ..., ...))

### ISSUE THREAD

**Issue by MorrisJobke** [scrutinizer-ci/php-analyzer#541]
I guess the variable $mountPoint isn't detected, because it's created in the list() inside of for

```
for( $mountPoints as list($bla, $mountPoint)) {
   $mountPoint->doSomething(); // this is detected as not a valid variable
}
```

see https://scrutinizer-ci.com/g/owncloud/core/inspections/bb643d41-df38-48e4-9def-d6dd8d3782e8/issues/files/lib/private/files/node/folder.php?status=new&selectedAuthors%5B0%5D=pvince81%40owncloud.com&orderField=path&order=asc&issueId=83fe917d-2e8a-4a80-bf46-0794a834aa75

**Comment by stof** [comment]
duplicate of #446
